### PR TITLE
feature/versioned_ts

### PR DESCRIPTION
**Pull request by devinrei**

### CODE REVIEW

File: cwms-radar-client/src/main/java/mil/army/usace/hec/cwms/radar/client/controllers/TimeSeriesEndpointInput.java [hunk 1]
```diff
@@ -36,7 +36,7 @@
 import mil.army.usace.hec.cwms.http.client.HttpRequestBuilder;
 import mil.army.usace.hec.cwms.radar.client.model.TimeSeries;
 
-public final class TimeSeriesEndpointInput {
+public final class      TimeSeriesEndpointInput {
```
Comment (adamkorynta) [inline cwms-radar-client/src/main/java/mil/army/usace/hec/cwms/radar/client/controllers/TimeSeriesEndpointInput.java:39]: ?

Comment (devinrei) [reply cwms-radar-client/src/main/java/mil/army/usace/hec/cwms/radar/client/controllers/TimeSeriesEndpointInput.java]: Fixed

File: cwms-radar-client/src/test/resources/radar/v2/json/timeseries_max_agg.json [hunk 2]
```diff
@@ -0,0 +1,52 @@
+{
+  "begin": "2018-01-05T00:00:00+0000[Z]",
+  "end": "2018-02-05T00:00:00+0000[Z]",
+  "version-date": "2024-03-11T00:00:00+0000[Z]",
```
Comment (adamkorynta) [inline cwms-radar-client/src/test/resources/radar/v2/json/timeseries_max_agg.json:4]: max aggregate shouldn't have a date

File: cwms-radar-client/src/test/java/mil/army/usace/hec/cwms/radar/client/controllers/TestTimeSeriesController.java [hunk 3]
```diff
@@ -99,6 +103,63 @@ void testRetrieveTimeSeries() throws IOException {
...
+                .begin(start)
+                .end(end)
+                .page(null);
+        TimeSeries timeSeries = new TimeSeriesController().retrieveTimeSeries(buildConnectionInfo(), input);
+        assertEquals(3, timeSeries.getValues().size());
+        assertEquals(3, timeSeries.getTotal());
+        assertEquals("SWT", timeSeries.getOfficeId());
+        assertEquals("m", timeSeries.getUnits());
+        assertEquals(Duration.ZERO, timeSeries.getInterval());
+        assertEquals(VersionType.MAX_AGGREGATE, timeSeries.getVersionType());
```
Comment (adamkorynta) [inline cwms-radar-client/src/test/java/mil/army/usace/hec/cwms/radar/client/controllers/TestTimeSeriesController.java:155]: should also assert that the date is null

File: cwms-radar-client/src/test/resources/radar/v2/json/timeseries_max_agg.json (hunk 2 above)
Comment (devinrei) [reply cwms-radar-client/src/test/resources/radar/v2/json/timeseries_max_agg.json]: Fixed

File: cwms-radar-client/src/test/java/mil/army/usace/hec/cwms/radar/client/controllers/TestTimeSeriesController.java (hunk 3 above)
Comment (devinrei) [reply cwms-radar-client/src/test/java/mil/army/usace/hec/cwms/radar/client/controllers/TestTimeSeriesController.java:154]: Fixed

File: cwms-radar-model/src/main/java/mil/army/usace/hec/cwms/radar/client/model/enums/VersionType.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
+package mil.army.usace.hec.cwms.radar.client.model.enums;
+
+public enum VersionType {
+    MAX_AGGREGATE,
+    SINGLE_VERSION,
+    UNVERSIONED;
+
+    public static VersionType versionTypeFor(String versionType) {
```
Comment (adamkorynta) [inline cwms-radar-model/src/main/java/mil/army/usace/hec/cwms/radar/client/model/enums/VersionType.java:8]: this method looks unused

**Review by adamkorynta — APPROVED** [review]

**Pull request merged**